### PR TITLE
Fixed:Buckets, Enchanting tables crafting, Use of deprecated dyes remove, Stone Brick

### PR DESCRIPTION
**Pull request by imjack**
Buckets - Fixed so they don't randomly disappear among other bugs that happens with buckets. AKA #1266
Enchanting tables - Fixed the crafting recipe for it
Dyes - Certain dyes used deprecated values so I replaced them.
Stone Brick - they gave the wrong amount

### CODE REVIEW

File: src/main/java/cn/nukkit/inventory/BaseInventory.java [hunk 1]
```diff
@@ -275,7 +275,7 @@ public boolean canAddItem(Item item) {
             }
 
             for (Item slot : new ArrayList<>(itemSlots)) {
-                if (slot.equals(item) && item.getCount() < item.getMaxStackSize()) {
+                if (slot.equals(item) && item.getCount() < item.getMaxStackSize() && slot.getDamage() == item.getDamage()) {
```
Comment (Pub4Game) [inline src/main/java/cn/nukkit/inventory/BaseInventory.java]: https://github.com/Nukkit/Nukkit/blob/master/src/main/java/cn/nukkit/item/Item.java#L1825-L1827

Already

File: src/main/java/cn/nukkit/item/ItemBucket.java [hunk 2]
```diff
@@ -86,13 +91,18 @@ public boolean onActivate(Level level, Player player, Block block, Block target,
...
                 player.getLevel().setBlock(block, targetBlock, true, true);
                 if (player.isSurvival()) {
-                    this.setCount(this.getCount() - 1);
-                    player.getInventory().setItemInHand(this);
-                    player.getInventory().addItem(ev.getItem());
+                    if (this.getCount() == 1) {
+                        player.getInventory().setItemInHand(ev.getItem());
+                    } else {
+                        this.setCount(this.getCount() - 1);
+                        player.getInventory().setItemInHand(this);
```
Comment (Pub4Game) [inline src/main/java/cn/nukkit/item/ItemBucket.java]: It is totally useless

**Comment by imjack** [comment]
@Pub4Game  should be better.. I hope

**Comment by MagicDroidX** [comment]
Please format your codes. :)

**Comment by imjack** [comment]
What formatting is wrong @MagicDroidX

**Comment by Creeperface01** [comment]
Code formating. Use reformat function to fix it

**Comment by MagicDroidX** [comment]
Conflicted. :(

**Comment by Pub4Game** [comment]
@MagicDroidX You break 1.0.0 branch... Thanks :(

**Pull request closed without merging**